server/mqtt.go

```diff
@@ -1960,6 +1964,7 @@ func (as *mqttAccountSessionManager) processRetainedMsg(_ *subscription, c *clie
 	as.handleRetainedMsg(rm.Subject, &mqttRetainedMsgRef{sseq: seq}, rm, false)
 
 	// If we were recovering (lastSeq > 0), then check if we are done.
+	// TODO <>/<> does this work with ReplayImmediate?
```

Comment:
@derekcollison @kozlovic Wouldn't the sequence check here potentially fire too early? Does `ReplayInstant` maintain the sequence order? Was not sure from the godoc. (there is no change in this PR, just curious)

https://github.com/nats-io/nats-server/blob/cca23cdacc66411905fe80fd7d17a9ccc8443401/server/mqtt.go#L1462

Reply:
ReplayInstant just means that the messages are resent without preserving the gap in time that there may be between original ingress messages. So the order should be preserved.